Userland/Libraries/LibGUI/Desktop.cpp

```diff
@@ -55,22 +55,26 @@ void Desktop::set_wallpaper_mode(StringView mode)
     WindowServerConnection::the().async_set_wallpaper_mode(mode);
 }
 
-bool Desktop::set_wallpaper(StringView path, bool save_config)
+String Desktop::wallpaper_path() const
 {
-    WindowServerConnection::the().async_set_wallpaper(path);
+    return Config::read_string("WindowManager", "Background", "Wallpaper");
+}
+
+RefPtr<Gfx::Bitmap> Desktop::wallpaper_bitmap() const
+{
+    return WindowServerConnection::the().get_wallpaper().bitmap();
+}
+
+bool Desktop::set_wallpaper(RefPtr<Gfx::Bitmap> wallpaper_bitmap, Optional<String> path)
+{
+    WindowServerConnection::the().async_set_wallpaper(wallpaper_bitmap->to_shareable_bitmap());
     auto ret_val = WindowServerConnection::the().wait_for_specific_message<Messages::WindowClient::SetWallpaperFinished>()->success();
 
-    if (ret_val && save_config) {
-        dbgln("Saving wallpaper path '{}' to ConfigServer", path);
-        Config::write_string("WindowManager", "Background", "Wallpaper", path);
+    if (ret_val && path.has_value()) {
+        dbgln("Saving wallpaper path '{}' to ConfigServer", *path);
+        Config::write_string("WindowManager", "Background", "Wallpaper", *path);
     }
```

Comment:
Would it not be better to have this function only write to the Config and then for the Config listener to be the only one to ever actually set the wallpaper?